src/livestreamer/plugins/media_ccc_de.py

```diff
@@ -0,0 +1,220 @@
+                                     "(?P<room>.*)"
+                                     "\/.*")
+
+def get_event_id(url):
+    """Extract event id from talk html page.
+
+    Raises :exc:`PluginError` on failure.
+
+    :param url: talk URL
+
+    """
+    match = re.search(r"{event_id:\s(?P<event_id>\d+),.*}", get_page(url))
+
+    try:
+        event_id = int(match.group('event_id'))
+    except:
+        raise PluginError("Failed to get event id from URL.")
+
+    return event_id
+
+def get_page(url):
+    """Fetch page for given URL.
+
+    :param url: URL to fetch
+
+    """
+    page = requests.get(url)
+
+    return page.text
+
+def create_json_object(json_string):
+    """Loads json string and returned Python object.
+
+    Raises :exc:`PluginError` on failure.
+
+    :param json_string: json as string
+
+    """
+    try:
+        json_object = json.loads(json_string)
+    except:
+        raise PluginError("Could not parse json from API.")
+
+    return json_object
+
+def parse_media_json(json_string):
+    """Expose available
+
+    :param json_string: json as string
+
+    """
+    json_object = create_json_object(json_string)
+
+    recordings = {}
+    for recording in json_object['recordings']:
+        match       = re.search(r".*\/(?P<format>.*)", recording['mime_type'])
+        file_format = match.group('format')
+
+        if recording['mime_type'] == 'vnd.voc/mp4-web' or\
+            recording['display_mime_type'] == 'video/webm':
+            continue
+        elif recording['mime_type'] == 'vnd.voc/h264-hd':
+            name = "1080p"
+        elif recording['mime_type'] == 'vnd.voc/h264-lq':
+            name = "420p"
+        elif re.match(r"audio", recording['display_mime_type']):
+            name = "audio_%s" % file_format
+        else:
+            if recording['hd'] == 'True':
+                name = "1080p"
+            else:
+                name = "420p"
+
+        recordings[name] = recording['recording_url']
+
+    return recordings
+
+def parse_streaming_media_json(json_string, room_from_url):
+    """Filter all availabe live streams for given json and room name.
+
+    API-Doku: https://github.com/voc/streaming-website#json-api
+
+    :param json_string: json as string
+    :param room_from_url:
+
+    """
+    json_object = create_json_object(json_string)
+
+    streams = {}
+    for group in json_object:
+        for room in group['rooms']:
+            # only consider to requested room
+            match = _url_streaming_media_re.match(room['link'])
+            if not match.group('room') == room_from_url:
+                continue
+
+            for stream in room['streams']:
+                # get stream language
+                if stream['isTranslated'] == False:
+                    language = 'native'
+                else:
+                    language = 'translated'
+
+                # get available hls stream urls
+                if 'hls' in stream['urls'].keys():
```

Comment:
No need for `.keys()`, you can do `if 'hls' in stream['urls']` instead, or to avoid repeating the nested structure:
```python
hls_stream = stream['urls'].get('hls')
if hls_stream:
    stream_url = hls_stream['url']
    ...
```